Enable make check (unit tests only) and check for warnings in travis

dlespiau opened the pull request:
Another step towards better CI:

- Get rid of warnings and make sure we don't introduce any later on
- Run make check

What's still to do is documented in https://github.com/01org/cc-oci-runtime/issues/69

grahamwhaley commented on src/commands/create.c:
```diff
@@ -22,6 +22,10 @@
 
+/* ignore -pedantic to cast handle_option_console, a function pointer, to a
+ * void* */
+#pragma GCC diagnostic push
```
I'd like to hear @jodh-intel opinion on the pragmas. Great that you've nullified the warnings, but in-code GCC specific pragmas are not generally to my liking :-)  If James is OK, then I think we'll probably be fine.

jodh-intel replied:
@grahamwhaley - I agree, not generally a fan of pragmas, but aside from changing `-std=`, I don't know how we can suppress these warnings.

Other thoughts on pragmas / options to suppress:

- will static analysis tools understand them?
- will other compilers understand them?

dlespiau replied:
I guess the answers to those questions will have to come with time. Worse than can happen is being ignored though, maybe with a warning.

The good news is that it's now also working with clang.

grahamwhaley commented on tests/hypervisor_test.c:
```diff
@@ -524,12 +524,12 @@ START_TEST(test_cc_oci_vm_args_get) {
 
 	/* recreate the args file */
-	ret = g_file_set_contents (args_file, "hello\# comment\n", -1, NULL);
+	ret = g_file_set_contents (args_file, "hello\\# comment\n", -1, NULL);
```
Good catch - but then this hints to me that the test was not really working or maybe testing what it should... @devimc can you have a think about that?

dlespiau commented:
Don't really want to open the pandora box, but I'd drop pedantic for -Wall and either all or some of the -Wextra warnings. I have yet to find an actually bug -pedantic helped catching and considering we're only targeting linux and x86_64, seems overkill to me.

grahamwhaley commented:
Nice series, thanks for doing that. Overall it looks good to me. Just a couple of minor comments.

jodh-intel commented:
To be clear, I'm +1 too. We might want to revisit the use of `-pedantic` at some point, but for now the ability to have Travis running the tests is more important IMHO. Added to which the pragma is commented and they are easy to `grep` :)

grahamwhaley commented:
OK, I was about to merge this and let @devimc pick up on the \\# test item above separately - but we now have merge conflicts. @dlespiau , if you could fix that then we'll merge.  Thanks.

dlespiau commented:
A few changes:

- Rebased on top of master
- Added a couple more commits to compile warning free with clang
- Changed the pragma to ignore "-Wpedantic" instead of "-pedantic" as the latter wasn't working with clang (I usually include a change log in the commit message)

dlespiau commented:
Ok, passing again. This time it was and interaction with the parallel build (-j5) that just got added, patch has the details.
